src/py/flwr/client/app.py

```diff
@@ -43,13 +46,34 @@
 from flwr.common.message import Error
 from flwr.common.retry_invoker import RetryInvoker, RetryState, exponential
 from flwr.common.typing import Run, UserConfig
+from flwr.server.superlink.state.utils import generate_rand_int_from_bytes
 
 from .grpc_adapter_client.connection import grpc_adapter
 from .grpc_client.connection import grpc_connection
 from .grpc_rere_client.connection import grpc_request_response
 from .message_handler.message_handler import handle_control_message
 from .node_state import NodeState
 from .numpy_client import NumPyClient
+from .process.process import run_clientappio_api_grpc
+
+CLIENTAPPIO_PORT = 9094
+ADDRESS_CLIENTAPPIO_API_GRPC_RERE = "0.0.0.0"
+
+
+def find_free_port(start_port):
+    port = start_port
+    while True:
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+            try:
+                s.bind(("", port))
+                s.close()
+                return port
+            except OSError:
+                port += 1
+
+
+CLIENTAPPIO_PORT_FREE = find_free_port(CLIENTAPPIO_PORT)
+ADDRESS_CLIENTAPPIO_API_GRPC_RERE += f":{CLIENTAPPIO_PORT_FREE}"
```

Comment:
What's the purpose of this? I don't think we should have dynamic server port assignments anywhere. I think it would be good to start a docs page where we list all default ports various Flower server processes use and just put the port into that list and have a fixed default.

Reply:
This mode is for users during development, where they start several `SuperNode`s on their local machine before moving to production. Since each `SuperNode` needs to have their own dedicated address, I added this utility script to ensure they use the next free port.

How about I move this to `supernode/connection.py`. I can update this such that we prompt users when the specified port `9094` is unavailable and suggest the next available port automatically?

Or is there a better way? Wdyt @danieljanes @tanertopal?

💯 on the list. I'll create another PR.

Reply:
Removed dynamic port allocation.